crates/bevy_ecs/src/reflect/component.rs

```diff
@@ -290,3 +284,19 @@ impl<C: Component + Reflect + FromWorld> FromType<C> for ReflectComponent {
         })
     }
 }
+
+fn from_reflect_or_world<T: FromReflect>(reflected: &dyn Reflect, world: &mut World) -> T {
+    if let Some(value) = T::from_reflect(reflected) {
+        return value;
+    }
+
+    let reflect_from_world = world
+        .resource::<AppTypeRegistry>()
+        .read()
+        .get_type_data::<ReflectFromWorld>(TypeId::of::<T>())
+        .unwrap()
+        .clone();
```

Comment:
I would prefer to see a `Option<ReflectFromWorld>` passed to `ReflectComponent.copy`.

There is a subtile issue with this in the `reflect/entity_commands.rs` module. Where users can pass their own `TypeRegistry`.

Reply:
Should this also apply to `ReflectComponent.insert` and `ReflectComponent.apply_or_insert`?

Reply:
yeah. To me (as a consumer of the API) I think having the `ReflectFromWorld` passed explicitly as an `Option` is the better pattern. Maybe controversial though. But I don't like the implicit use & panic on the from_world.